### ISSUE TITLE
Visualization does not show up in the distribution version. [CARROT-358]

### ISSUE THREAD

**Issue by dweiss** [carrot2/carrot2#563]
The visualization plugin is included in the binary distribution, but does not show up in the list of views. Interestingly, the pde error view does not show any exceptions and the plugin is not loaded in the configuration. Why?

Ula, would you take a look at this? I would appreciate.

---
Issue: CARROT-358 (migrated from JIRA), created by Dawid Weiss (@dweiss), resolved Aug 09 2008
Linked issues:
 - CARROT-361

**Comment by dweiss** [comment]
Comment by Urszula Krukar (migrated from JIRA)

Few plugins were missing, I've added them already, It still does not show up.

As I understand, dependencies of the org.mortbay.jetty plugin are not resolved. This plugins needs these packages to be available:
javax.net.ssl,
javax.security.cert,
javax.servlet;version="[2.4.0,2.5.0)",
javax.servlet.http;version="[2.4.0,2.5.0)",
javax.servlet.resources;version="[2.4.0,2.5.0)",
javax.xml.parsers,
org.apache. commons.logging;version="[1.0.0,2.0.0)",
org.xml.sax,
org.xml.sax.helpers

I am not exactly sure, where to take them all from.

PS. Mortbay plugin used in the Workbench is taken from Eclipse target platform, not from lib directory. This mortbay project is not a plugin (wasn't converted). This is intentional?

**Comment by dweiss** [comment]
Comment by Dawid Weiss (@dweiss) (migrated from JIRA)

All these plugins should actually come with the target platform... or maybe rather: they are part of the default Eclipse (not necessarily JDT). I used standard servlet extensions used by (among others) the help component. I am surprised they are not part of the default platform.

I will take a look at it during the weekend if you can't figure it out before then.

As for the Jetty plugin – our lib/ folder contains a project that is used by the Webapp. Workbench has a dependency on Jetty because Jetty is the default implementation of the http bundle. This is all right, in other words.

**Comment by dweiss** [comment]
Comment by Urszula Krukar (migrated from JIRA)

All these plugins come as a part of  `org.eclipse.platform` feature, but Workbench depends on `org.eclipse.rcp` feature only. As a "quick fix" I added `platform` feature as a part of `Workbench.product` instead of `rcp`. It solved the problem, but many unnecessary menu items, views, actions etc. were added to the Workbench in the process.

Two things you can do now:
- manually hide all those unnecessary items, actions etc.
- fins only those plugins from the `platform` that are necessary for visualization to work, and add them to `org.carrot2.workbench` feature.

**Comment by dweiss** [comment]
Comment by Dawid Weiss (@dweiss) (migrated from JIRA)

The solution would be to build the workbench not based on features (or not only on features), but on plugins. Then we won't have to deal with the extra stuff that comes with each feature – the dependencies should be handled semi-automatically with the help of the PDE.

**Comment by dweiss** [comment]
Comment by Urszula Krukar (migrated from JIRA)

Fixed in trunk.

What was the problem:
- plugins missing:
  - `org.eclipse.equinox.http.jetty`
  - `org.eclipse.equinox.http.registry`
  - `org.eclipse.equinox.http.servlet`
  - `org.mortbay.jetty`
- `org.mortbay.jetty` has this dependency among others:
   `org.apache. commons.logging;version="[1.0.0,2.0.0)"` and it was not resolved!! We added `org.apache.commons` plugin to Workbench, but it did not specify packages versions. I added appriopriate version to `org.apache.commons.logging` package, and now it works :D
